### ISSUE TITLE
Consider offering Log4J2 dependency

### ISSUE THREAD

**Issue by philwebb** [spring-io/start.spring.io#183]
See [this Spring Boot issue for background](https://github.com/spring-projects/spring-boot/issues/16864).

Although it's unlikely that we're going to be able to consider switching the default logging system until Spring Boot 3.x, it might be nice if people could pick Log4J2 from start.spring.io if they wish. Picking the dependency would need to add Log4J2 and exclude Logback.

**Comment by wilkinsona** [comment]
If we were to offer an option for an alternative logging framework, I'd expect it to be something that we did after offering alternative web servers. From what I've seen, those are customized far more often than the logging framework.

If we started offering alternative, I think things may get confusing. For example, if there's a checkbox for Log4j2 but there isn't one for Logback, a user may think that they won't get any logging framework at all unless they choose Log4j2. We could then also offer a checkbox for Logback but I think that raises more questions than it answers:

- What should happen if a user selects both?
- What should happen if a user doesn't select either of them?

We considered this sort of thing with JUnit 5 and decided that the best option was to remain opinionated and align with Boot's defaults. I think we should do the same here as well and wait for a change in Boot (should one ever be made).
